### PR TITLE
Change old admin page (based on oauth and multi-tenent model) to RSI model with data load support

### PR DESCRIPTION
**Pull request by shifucun**
On server start, can set an environment variable "ADMIN_SECRET". When user perform actions in this admin page, they need to input the secret. This is a naive way for authentication before we have a full solution.

image

### CODE REVIEW

File: server/routes/admin/html.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+          "status": "success",
+          "stdout": result.stdout.strip().splitlines()
+      })
+    except subprocess.CalledProcessError as cpe:
+      return jsonify({
+          "status": "failure",
+          "error": cpe.stderr.strip().splitlines()
+      }), 500
+    except Exception as e:
+      return jsonify({"status": "error", "message": str(e)}), 500
```
[the remark this answers is not in the document]
Comment (juliawu) [reply server/routes/admin/html.py:77]: Perhaps we can log the error so we can debug internally and instead return a more generic message to the user?

**Review by juliawu — APPROVED** [review]
Nice! Just some quick thoughts on the alerts.

File: server/routes/admin/html.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (shifucun) [reply server/routes/admin/html.py:77]: The log here is for the custom DC dev uses, so the errors are desired. 

Moving forward, we should put more stuff in a status.json file for the page rendering purpose.

File: server/routes/admin/html.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+      "-X",
+      "POST",
+      "localhost:8081/import",
+      "-d",
+      f'{{"data_path": "{output_dir}"}}',
+  ]
+  output = []
+  for command, cwd in [(command1, "import/simple"), (command2, ".")]:
+    try:
+      result = subprocess.run(command,
```
[the remark this answers is not in the document]
Comment (shifucun) [reply server/routes/admin/html.py:62]: Added some check for command, but this warning would still persist.

**Review by shifucun — COMMENTED** [review]
thanks for review!

File: import
Comment (chejennifer) [inline import:1]: did you mean to update this and the import submodules?

File: static/js/admin/app.ts
Comment (chejennifer) [inline static/js/admin/app.ts:1]: nit: maybe make this `main.ts` to be consistent with the other pages where `main.ts` is the file with the "window.onload"

File: static/js/admin/page.tsx [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+ * Admin Page.
+ */
+
+import axios from "axios";
+import _ from "lodash";
+import React, { useState } from "react";
+import { Button } from "reactstrap";
+
+export function Page(): JSX.Element {
+  const [loading, setLoading] = useState(Boolean);
```
Comment (chejennifer) [inline static/js/admin/page.tsx:27]: is this a new syntax for `useState`? Because shouldn't the state type be declared like `useState<boolean>()`

**Review by chejennifer — COMMENTED** [review]
cool!

File: nl_server/requirements.txt [hunk 4]
```diff
@@ -1,20 +1,20 @@
 datasets==2.12.0
```
Comment (shifucun) [inline nl_server/requirements.txt:1]: sorted the packages

File: server/requirements.txt [hunk 5]
```diff
@@ -1,25 +1,26 @@
 absl-py==1.4.0
```
Comment (shifucun) [inline server/requirements.txt:1]: sorted the pip packages here

File: static/js/admin/page.tsx (hunk 3 above)
Comment (shifucun) [reply static/js/admin/page.tsx]: Good catch, there were totally off!

**Pull request merged**